### PR TITLE
docs: FAQ by categories as proposed by @anarcat in #1802

### PR DESCRIPTION
**Pull request by enkore**
docs: FAQ by categories as proposed by @anarcat in #1802

Usage & Limitations
Security
Common issues
Miscellaneous (only three items, two fork-related)

I immediately noticed that this makes it *much* easier to find FAQ items -- the list is just too long without any kind of hierarchy / order.

Note: This does not change/invalidate any links to FAQ items.

### CODE REVIEW

File: docs/faq.rst [hunk 1]
```diff
@@ -93,7 +96,7 @@ Which file types, attributes, etc. are *not* preserved?
       backed up as (deduplicated and compressed) runs of zero bytes.
       Archive extraction has optional support to extract all-zero chunks as
       holes in a sparse file.
-    * filesystem specific attributes, like ext4 immutable bit, see :issue:`618`.
+    * Some filesystem specific attributes, like btrfs NOCOW, see :ref:`platforms`.
```
Comment (ThomasWaldmann) [inline docs/faq.rst]: what was the problem with giving ext4 immutable bit as example and pointing to the issue?

File: docs/faq.rst [hunk 2]
```diff
@@ -120,6 +122,70 @@ decision about whether you want to use compression, which algorithm
 and which level you want to use. This is why compression defaults to
 none.
 
+If a backup stops mid-way, does the already-backed-up data stay there?
+----------------------------------------------------------------------
+
+Yes, |project_name| supports resuming backups.
```
Comment (ThomasWaldmann) [inline docs/faq.rst]: Reminds me I need to rephrase that. In fact, we never resume a backup, we always do full backups from the beginning.
It just feels like resuming...

File: docs/faq.rst (hunk 1 above)
Comment (enkore) [inline docs/faq.rst]: ext4 immutable has been supported for a while

File: docs/faq.rst (hunk 2 above)
Comment (enkore) [inline docs/faq.rst]: It's a bit like "Does Borg do incremental backups?" "Well, uh... it certainly feels like it! But <complex explanation>" :)

File: docs/faq.rst (hunk 1 above)
Comment (ThomasWaldmann) [inline docs/faq.rst]: oh, ok. :)

**Pull request merged**